src/autopas/selectors/tuningStrategy/LiveInfo.h

```diff
@@ -0,0 +1,325 @@
+        const auto diffFromAvg = avgParticlesPerCell - static_cast<int>(particlesInBin);
+        maxDiffLambda = std::max(diffFromAvg, maxDiffLambda);
+        sumStddevLambda += diffFromAvg * diffFromAvg;
+      }
+      estimatedNumNeighborInteractionsLambda /= 2;
+
+      return std::tuple{estimatedNumNeighborInteractionsLambda,
+                        maxDiffLambda,
+                        sumStddevLambda,
+                        numEmptyCellsLambda,
+                        maxParticlesPerCellLambda,
+                        minParticlesPerCellLambda};
+    }();
+
+    infos["numEmptyCells"] = numEmptyCells;
+    infos["maxParticlesPerCell"] = maxParticlesPerCell;
+    infos["minParticlesPerCell"] = minParticlesPerCell;
+    infos["particlesPerCellStdDev"] =
+        std::sqrt(sumStddev) / static_cast<double>(particleBins.size() - 1) / avgParticlesPerCell;
+    infos["avgParticlesPerCell"] = avgParticlesPerCell;
+    infos["estimatedNumNeighborInteractions"] = static_cast<unsigned long>(estimatedNumNeighborInteractions);
+
+    const double sumStddevBlurred = [&]() {
+      double res = 0.;
+      for (auto numParticlesInBin : particleBinsBlurred) {
+        auto diff = avgParticlesPerBlurredCell - static_cast<int>(numParticlesInBin);
+        res += diff * diff;
+      }
+      return res;
+    }();
+    infos["particlesPerBlurredCellStdDev"] =
+        std::sqrt(sumStddevBlurred) / static_cast<double>(particleBinsBlurred.size()) / avgParticlesPerBlurredCell;
+
+    infos["threadCount"] = static_cast<size_t>(autopas::autopas_get_max_threads());
+
+    constexpr size_t particleSizeNeededByFunctor = calculateParticleSizeNeededByFunctor<Particle, PairwiseFunctor>(
+        std::make_index_sequence<PairwiseFunctor::getNeededAttr().size()>());
+    infos["particleSizeNeededByFunctor"] = particleSizeNeededByFunctor;
+  }
+
+  /**
+   * Returns a map of all infos.
+   * @return A map of all infos.
+   */
+  [[nodiscard]] const auto &get() const { return infos; }
+
+  /**
+   * Creates a string containing all live info gathered.
+   * @return A string containing all live info gathered.
+   */
+  [[nodiscard]] std::string toString() const {
+    auto typeToString = [](auto type) {
+      if constexpr (std::is_same_v<decltype(type), bool> or std::is_same_v<decltype(type), double> or
+                    std::is_same_v<decltype(type), size_t>) {
+        return std::to_string(type);
+      } else if constexpr (std::is_base_of_v<Option<decltype(type)>, decltype(type)>) {
+        return type.to_string();
+      }
+      return std::string{"fail"};
+    };
+    auto toString = [&](const auto &pair) { return pair.first + "=" + std::visit(typeToString, pair.second); };
+    std::string res{"Live Info: "};
+    if (not infos.empty()) {
+      res += std::accumulate(std::next(infos.begin()), infos.end(), toString(*infos.begin()),
+                             [&](std::string s, const auto &elem) { return std::move(s) + " " + toString(elem); });
```

Comment:
@HobbyProgrammer why `std::next()`? Doesn't this mean that the fist entry is skipped in the output?

Reply:
The first entry is passed initial value for the accumulation (`toString(*infos.begin())`). We don't want it twice in the output, so we skip the first element in the passed iterator. Passing an empty string as initializer would be nicer, but then we would have a space in the beginning, so we can't do it that way.

Reply:
Ah I missed the fact that you chose the first element as the initializer. Makes sense now. I'll add a comment.